Generalizations for notifications

alexjgriffith opened the pull request:
Generalizations of `mastodon-tl--toot` and `mastodon-tl--byline` that allow them to be used to render notifications.

Three new arguments are added to `mastodon-tl--toot`
1. body: appears above the byline
2. author-byline, a functions that defines how the author should be rendered and that defaults to `mastodon-tl--byline-author`
2. action-byline, a functions that defines how the action should be rendered. Actions include boosting, following, mentioned etc. It defaults to `mastodon-tl--byline-boosted`

I also removed the redundant removal of newlines in `mastodon-tl--timeline`

alexjgriffith committed:
Made generalizations to toot and bylines to enable notifications.

hdurer commented on lisp/mastodon-tl.el:
```diff
@@ -324,8 +324,14 @@ E.g. this could return something like \"1 min ago\", \"yesterday\", etc.
+  "Generate byline for TOOT.
+
+AUTHOR-BYLINE is function for adding the author portion of 
+the byline that takes one variable. By default it is `mastodon-tl--byline-author'
```
`By default`is now meaningless.

hdurer commented on lisp/mastodon-tl.el:
```diff
@@ -483,26 +489,39 @@ message is a link which unhides/hides the main body."
 
-(defun mastodon-tl--toot (toot)
-  "Display TOOT content and byline."
+(defun mastodon-tl--insert (toot body author-byline action-byline)
```
`mastodon-tl--insert` is too generic  name.   
If this going to be more generic than just inserting toots, we should come up with a good name for this generic "something" and then also rename the `toot` argument and fix the doc comment to not say `toot`.

hdurer commented on lisp/mastodon-tl.el:
```diff
@@ -483,26 +489,39 @@ message is a link which unhides/hides the main body."
+  "Display toot content and byline.
+
+BODY will form the section of the toot above the byline.
+AUTHOR-BYLINE and ACTION-BYLINE are optional functions that are passed to
```
Remove the `optional` bits.

alexjgriffith committed:
Added generalizations needed for adding notifications #37

- Made generalizations to toot and bylines to enable notifications.
- Fixed tl tests to account for the new function arguments.
- fixed whitespace
